### ISSUE TITLE
failed to update coverage via rest api

### ISSUE THREAD

**Issue by zhiyuli** [hydroshare/hydroshare#3027]
This rest call is failing against 1.16.1 and 1.17.  
I got a 400 error on the client, but I couldnt find any error messages in log files on a testing server that seems to be related.

 metadata = {               
                 "coverages": [
                                                            {"type": "period", "value": {"start": "01/01/2000", "end": "12/12/2010"}}
                                                      ],                         
                          }
 science_metadata_json = hs.updateScienceMetadata('ID OF RESOURCE GOES HERE', metadata=metadata)

**Comment by zhiyuli** [comment]
@sblack-usu @pkdash @horsburgh @dtarb 
I found the failure is caused by this line. So it is by design? 
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L3733

I was able to manually update resource-level coverage for a composite resource on UI. But currently, REST API is prohibiting it. 

Also, the specific error message didn't bubble up to the client, which caused troubles in debugging unless going through the source code.

**Comment by zhiyuli** [comment]
Also this file has more than 4000 lines codes.....it even takes github 1-2 sec to load for the first time...
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L3733

**Comment by pkdash** [comment]
@zhiyuli 

> I found the failure is caused by this line. So it is by design?
https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L3733

The above update() function is used by REST API for bulk (updating multiple metadata elements)metadata update. At the time of implementation it was by design not to allow manual update of coverage for composite resource as the coverage gets set by the file types contained by the resource. Later in phase-2 of composite resource, this was changed and now we allow manual update of coverage for composite resource. The code line in reference needs to be fixed.

**Comment by sblack-usu** [comment]
@zhiyuli - we could likely squeeze this into the enxt release if you wanted to make the change.

**Comment by zhiyuli** [comment]
@sblack-usu   yeah. It seems to be a small change and  I would like to look into this. I need some guidance on PR policy though as I heard it has changed recently. Will bother you on slack.

**Comment by zhiyuli** [comment]
@pkdash @horsburgh 
Thanks for the explanations. I just notice the "if statement" that prevent updating coverage also applies to TimeSeriesResource as well. Do we want to keep unchanged for TimeSeriesResource (while lifting the ban for CompositeResType)?

**Comment by horsburgh** [comment]
I think we should probably not spend time updating time series resource, but rather focus on migration of existing time series resources to composite and then full deprecation of the time series resource type.

**Comment by zhiyuli** [comment]
OK. I assume that means we should keep everything here related to
TimeSeriesResource unchanged. Will update my PR accordingly.

**Comment by sblack-usu** [comment]
PR https://github.com/hydroshare/hydroshare/pull/3031

**Comment by zhiyuli** [comment]
This is still failing for me on current www (1.18). Need to check what is going on....
